demo/src/main/java/com/demo/MainApp.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.demo;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.web.servlet.ServletComponentScan;
+
+@SpringBootApplication(scanBasePackages = {"com.demo", "com.google.fhir.gateway.plugin"})
+@ServletComponentScan(basePackages = "com.google.fhir.gateway")
+public class MainApp {
```

Comment:
I feel what you have added in `/demo` is very similar to what we have in `/exec` and I am thinking of dropping one. The main reason for having `/exec` was to have an executable proxy to quickly test while developing server/plugins. This "demo" app can serve the same purpose. The main difference is that we need to use a SNAPSHOT version (instead of `0.1.1`) for server and plugin deps. The other difference is that since `exec` shared the same parent POM, it was easier to make changes and run. But with this `demo` we can work around that by doing a `mvn install` first (for server/plugin).

Anyway, the goal of this comment is to only keep one of `demo` and `exec`, up to you which way you prefer. If you want to keep `demo`, please also add the minimum Spring test similar to [this](https://github.com/google/fhir-gateway/blob/5ab10af69814da6b1e6a8357d1f904ccf07b8343/exec/src/test/java/com/google/fhir/gateway/MainAppTest.java#L26).

Reply:
demo will only pull the dependency from maven artifact but exec is depending on the local code. 
Other than you are right it is same. 

Is there anyway I can make that configurable and just have a one of demo/exec ?

Reply:
In this case let me see how this can be made configurable rather than someone manually changing the version in the pom.xml.
This will be required for running the e2e tests with the latest server changes.

Reply:
This is already take care of in the exec folder and I don't need to do anything. 

Only additional thing is adding the scan package in the MainApp annotation. It is not required in the exec because the package names are same for exec and the server/plugin.

I have added the default scan path in the exec Main file and deleted the demo folder.
I think this should be enough with good documentation. Let me know if you think otherwise.